src/platform/monitoring/DowntimeNotification/containers/DowntimeNotification.jsx

```diff
@@ -0,0 +1,101 @@
+ * @property {function} getScheduledDowntime - [Provided by container] An action creator that retrieves the array of downtime from the API downtime endpoint.
+ * @property {boolean} isReady - [Provided by container] A flag for indicating whether the downtime array has been retrieved from the API and if the component can render.
+ * @property {Node} [loadingIndicator] - A React component that will be rendered while the request to the API for downtime information is pending.
+ * @property {function} [render] - A function that may be supplied for custom rendering, useful for customizing how downtime/downtime approaching is handled. Receives the derived status, downtimeWindow, downtimeMap, children as arguments.
+ * @module platorm/monitoring/DowntimeNotification
+ */
+class DowntimeNotification extends React.Component {
+
+  static propTypes = {
+    appTitle: PropTypes.string,
+    children: PropTypes.node,
+    content: PropTypes.node,
+    dependencies: PropTypes.arrayOf(PropTypes.oneOf(Object.values(services))).isRequired,
+    getScheduledDowntime: PropTypes.func.isRequired,
+    isReady: PropTypes.bool,
+    loadingIndicator: PropTypes.node,
+    render: PropTypes.func
+  };
+
+  componentDidMount() {
+    if (this.props.shouldSendRequest) this.props.getScheduledDowntime();
+  }
+
+  render() {
+    if (!this.props.isReady) {
+      return this.props.loadingIndicator || <LoadingIndicator message={`Checking the ${this.props.appTitle} status...`}/>;
+    }
+
+    const children = this.props.children || this.props.content;
+
+    if (this.props.render) {
+      return this.props.render({
+        externalServive: this.props.externalServive,
+        status: this.props.status,
+        startTime: this.props.startTime,
+        endTime: this.props.endTime
+      }, children);
+    }
+
+    if (this.props.status === serviceStatus.downtimeApproaching) {
+      return <DowntimeApproaching {...this.props}/>;
+    }
+
+    if (this.props.status === serviceStatus.down) {
+      return <Down {...this.props}/>;
+    }
+
+    return children;
+  }
+}
+
+const mapStateToProps = (state, ownProps) => {
+  const scheduledDowntime = state.scheduledDowntime;
+  const shouldSendRequest = !scheduledDowntime.isReady && !scheduledDowntime.isPending;
+  const isDowntimeApproachingModalDismissed = scheduledDowntime.dismissedDowntimeApproachingModals.includes(ownProps.appTitle);
+
+  let downtime = null;
+  if (scheduledDowntime.isReady) {
+    downtime = getSoonestDowntime(scheduledDowntime.serviceMap, ownProps.dependencies);
+  }
+
+  return {
+    shouldSendRequest,
+    isDowntimeApproachingModalDismissed,
+    ...scheduledDowntime,
+    ...downtime
+  };
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  return {
+    getScheduledDowntime() {
+      return dispatch(getScheduledDowntime());
+    },
+    dismissDowntimeApproachingModal() {
+      return dispatch(dismissDowntimeApproachingModal(ownProps.appTitle));
```

Comment:
Yeah I could see that. My thinking here was binding directly to ownProps so that if the action is passed down to the children the parameter will still be there without worrying about the extra prop. But that could also be done just as a method, so either way.